GliaWidgets/SwiftUI/Components/SnackBarView/SerialQueue.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+
+/// Represents queue with serial strategy. Not thread-safe.
+final class SerialQueue {
+    /// Adds operation to end of  serial queue
+    func addOpeation(_ op: Operation) {
+        let needToRun = queue.isEmpty
+        queue.append(op)
+        if needToRun {
+            run()
+        }
+    }
+
+    // MARK: - Private
+
+    private var queue = [Operation]()
+
+    private func run() {
+        guard queue.isEmpty == false else { return }
```

Comment:
could also do something like `let op = queue.first else { return }` no? Although this version is more expressive.

Reply:
correct, missed it - b46b99e4ed9a6e92634870cb36cbdddad983bc83